llmfoundry/data/finetuning/tasks.py

```diff
@@ -371,10 +373,19 @@ def __init__(self,
         )
 
         self.tokenizer = tokenizer
+        self.max_seq_len = max_seq_len
 
     # How to process a sample
     def __getitem__(self, idx: int) -> Dict[str, Any]:
         sample = super().__getitem__(idx)
+        if 'input_ids' in sample:
+            # already tokenized data
+            sample['input_ids'] = np.frombuffer(
+                sample['input_ids'],
+                dtype=np.int64)[:self.max_seq_len].tolist().copy()
```

Comment:
Wondering, what is the reason behind `np.int64`? why not `int32` or lower?

Reply:
Honestly, i don't know. I just copied from [here](https://github.com/mosaicml/llm-foundry/blob/105f76631d7f74edebc187f3427008ce4506ee85/llmfoundry/data/text_data.py#L170-L171) . 

I guess it's better to set the dtype explicitly when converting to `numpy.array` [here](https://github.com/mosaicml/llm-foundry/blob/105f76631d7f74edebc187f3427008ce4506ee85/scripts/data_prep/convert_finetuning_dataset.py#L258) , and make this two place use the same dtype.  Do you suggest to do that ?

Reply:
as discussed offline, lets make the change we discussed in this pr